pshtt/pshtt.py

```diff
@@ -1199,16 +1199,20 @@ def is_missing_intermediate_cert(domain):
 
 def is_hsts(domain):
     """
-    Domain has HSTS if its canonical HTTPS endpoint has HSTS.
+    Domain has HSTS if both https and httpswww endpoints have HSTS when live.
```

Comment:
The canonical endpoint was (I think) introduced by @h-m-f-t in an effort to make the HTTPS report results more digestible by the less technical.  I'd have to know his thoughts on this change before approving.

Reply:
I think @climber-girl 's point was that if both endpoints are live then both should have HSTS. I think [this section of RFC6797](https://tools.ietf.org/html/rfc6797#section-11.4.2) touches on what I mean although it is in the context of a web application and the includeSubDomains directive.

Reply:
I don't completely understand all of the reasons for only focusing on the canonical endpoint. It seems like we assume that that is where most users will end up, but plenty of others may use the other endpoint (if it is live), and attackers could take advantage of the other endpoint (via spear-phishing URLs and the like) if it is not sufficiently protected. 

`strictly_forces_https` already looks at both http endpoints, not just the canonical one, but `downgrades_https` currently only looks at the canonical https endpoint. If we switch this HSTS check to check both endpoints, then we should probably change some other checks to look at both endpoints (such as `downgrades_https` and maybe even follow-on `sslyze` checks).  Should most of these checks for both endpoints be in new "Strictly_Enforces" checks or should these be changed in the existing checks?